src/components/main-view/main-view.jsx

```diff
@@ -1,29 +1,106 @@
 import { useEffect, useState } from "react";
 import { MovieCard } from "../movie-card/movie-card";
 import { MovieView } from "../movie-view/movie-view";
+import { LoginView } from "../login-view/login-view";
+import { SignupView } from "../signup-view/signup-view";
+
 
 export const MainView = () => {
+  const storedUser = JSON.parse(localStorage.getItem("user"));
+  const storedToken = localStorage.getItem("token");
```

Comment:
I see you're trying to retrieve the stored user and stored token but not using them anywhere. So the user and token is always `null`
You will need to set the token and set user from here.